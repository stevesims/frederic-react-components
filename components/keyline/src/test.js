import React from 'react';
import { mount } from 'enzyme';
import { createMatchers } from 'jest-emotion';
import * as emotion from 'emotion';
import { GREY_1 } from 'govuk-colours';

import Component from '.';

expect.extend(createMatchers(emotion));

describe('Keyline', () => {
  let wrapper;

  it('renders with no props', () => {
    wrapper = mount(<Component />);
    expect(wrapper).toHaveStyleRule('border-bottom', `1px solid ${GREY_1}`);
  });

  it('renders children', () => {
    wrapper = mount(<Component>Sample text</Component>);
    expect(wrapper.find('KeylineWrapper').childAt(0).text()).toBe('Sample text');
  });
 
  it('renders as inlineBlock', () => {
    wrapper = mount(<Component inlineBlock />);
    expect(wrapper).toHaveStyleRule('display', 'inline-block'); 
  });
  
  it('renders with borderColor', () => {
    wrapper = mount(<Component borderColor="red" />);
    expect(wrapper).toHaveStyleRule('border-bottom', '1px solid red'); 
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
