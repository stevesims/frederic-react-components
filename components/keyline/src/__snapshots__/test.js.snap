// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keyline matches snapshot 1`] = `
.emotion-2 {
  margin-bottom: 15px;
}

@media only screen and (min-width:641px) {
  .emotion-2 {
    margin-bottom: 15px;
  }
}

.emotion-0 {
  border: 0;
  border-bottom: 1px solid red;
  display: block;
  font-family: "nta",Arial,sans-serif;
  margin-bottom: 15px;
}

@media only screen and (min-width:641px) {
  .emotion-0 {
    margin-bottom: 15px;
  }
}

<Styled(Keyline)
  borderColor="red"
>
  <Keyline
    borderColor="red"
    className="emotion-2"
  >
    <KeylineWrapper
      borderColor="red"
      className="emotion-2"
    >
      <span
        className="emotion-0 emotion-1"
      />
    </KeylineWrapper>
  </Keyline>
</Styled(Keyline)>
`;
